packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx

```diff
@@ -870,8 +869,14 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
     return graph;
   };
 
-  private _getCircleRadius = (xDataPoint: number, circleRadius: number, circleId: string): number => {
-    const { isCircleClicked, nearestCircleToHighlight, activePoint } = this.state;
+  private _getCircleRadius = (xDataPoint: number, circleRadius: number, circleId: string, legend: string): number => {
+    const { isCircleClicked, nearestCircleToHighlight, activePoint, selectedLegends } = this.state;
+
+    // Show the circle if no legends are selected or if the point's legend is in the selected legends
+    if (selectedLegends.length > 0 && !selectedLegends.includes(legend)) {
+      return 0;
+    }
```

Comment:
this is missing out on legend hovered condition.

Reply:
updated to use _legendHighlighted